src/jetstream.rs

```diff
@@ -57,7 +57,7 @@
 //!
 //! js.add_stream("my_stream")?;
 //!
-//! let consumer: nats::jetstream::ConsumerSubscription = js.add_consumer("my_stream", "my_consumer")?;
+//! let consumer: nats::jetstream::ConsumerSubscription = js.subscribe("my_stream", "my_consumer")?;
```

Comment:
Same, subscribe should always take a subject and we do the work underneath the covers.

However, to create a sub from a consumer that you bind to should be possible.

```rust
let sub = consumer.sub();
```

Reply:
Ended up settling on discarding `Consumer` completely.
Lets come back to a potential lower level interface once we have feature parity with the other clients.